assets/js/atomic/blocks/product-elements/price/edit.tsx

```diff
@@ -51,26 +52,80 @@ const PriceEdit = ( {
 	};
 	const isDescendentOfQueryLoop = Number.isFinite( context.queryId );
 
+	const { isDescendentOfSingleProductTemplate } = useSelect(
+		( select ) => {
+			const store = select( 'core/edit-site' );
+			const postId = store?.getEditedPostId();
+
+			return {
+				isDescendentOfSingleProductTemplate:
+					( postId === 'woocommerce/woocommerce//product-meta' ||
+						postId ===
+							'woocommerce/woocommerce//single-product' ) &&
+					! isDescendentOfQueryLoop,
+			};
+		},
+		[ isDescendentOfQueryLoop ]
+	);
+
 	useEffect(
-		() => setAttributes( { isDescendentOfQueryLoop } ),
-		[ setAttributes, isDescendentOfQueryLoop ]
+		() =>
+			setAttributes( {
+				isDescendentOfQueryLoop,
+				isDescendentOfSingleProductTemplate,
+			} ),
+		[
+			isDescendentOfQueryLoop,
+			isDescendentOfSingleProductTemplate,
+			setAttributes,
+		]
 	);
 
+	const showProductSelector =
+		! isDescendentOfQueryLoop && ! isDescendentOfSingleProductTemplate;
+
+	if ( ! showProductSelector ) {
+		return (
+			<>
+				<BlockControls>
+					<AlignmentToolbar
+						value={ attributes.textAlign }
+						onChange={ ( textAlign: AllowedAlignments ) => {
+							setAttributes( { textAlign } );
+						} }
+					/>
+				</BlockControls>
+				<div { ...blockProps }>
+					<Block { ...blockAttrs } />
+				</div>
+			</>
+		);
+	}
+
 	return (
-		<>
-			<BlockControls>
-				<AlignmentToolbar
-					value={ attributes.textAlign }
-					onChange={ ( textAlign: AllowedAlignments ) => {
-						setAttributes( { textAlign } );
-					} }
-				/>
-			</BlockControls>
-			<div { ...blockProps }>
+		<div { ...blockProps }>
+			<ProductSelector
+				productId={ attributes.productId }
+				setAttributes={ setAttributes }
+				icon={ BLOCK_ICON }
+				label={ BLOCK_TITLE }
+				description={ __(
+					'Choose a product to display its SKU.',
```

Comment:
I think instead of `SKU`, we have to use `Price` here? 
```suggestion
					'Choose a product to display its Price.',
```

Reply:
Good point! Fixed via a3d1ba37730ba796060e7b9b7e40594434ca2e20